plugin/storage/grpc/shared/grpc_client.go

```diff
@@ -77,10 +77,9 @@ func (c *grpcClient) GetTrace(ctx context.Context, traceID model.TraceID) (*mode
 	trace := model.Trace{}
 	for received, err := stream.Recv(); err != io.EOF; received, err = stream.Recv() {
 		if err != nil {
-			if e, ok := status.FromError(err); !ok {
```

Comment:
wasn't this if statement mostly correct, except the condition should be `ok` instead of `!ok`?

Reply:
Yup, that's true.  I prefer what I have pushed b/c it doesn't matter if `err` is an `error` or a `status.Status` if the message is "trace not found" then it will return the correct error.

We can do the minimal change if it's preferred though.

Reply:
the original version (aside from !ok) seems better, because your version is ignoring the error, so `s` may be undefined.

Reply:
good point.  I am relying on `status.FromError` to return a non-nil status.  I think this would the best construction

```
if s, _ := status.FromError(err); s != nil {
  // check if s is a trace not found error.
}
```

`ok` just indicates if `FromError` had to create a new `status.Status` or if it was able to retrieve one using the `GRPCStatus()` method.  For our purposes we don't care, we're just going to compare `s.Message()` to "trace not found" either way.